src/System.Management.Automation/engine/remoting/fanin/PriorityCollection.cs

```diff
@@ -154,15 +156,29 @@ internal void Add<T>(RemoteDataObject<T> data)
         /// </summary>
         internal void Clear()
         {
-            Dbg.Assert(null != _dataToBeSent, "Serialized streams are not initialized");
-            lock (_syncObjects[(int)DataPriorityType.PromptResponse])
+            /* Lock _syncObjects and _dataToBeSent as a set to ensure atomic cleanup. */
+            lock (_syncObject)
```

Comment:
Is it possible to lock on '_syncObjects' array itself without creating additional field "_syncObject" ?